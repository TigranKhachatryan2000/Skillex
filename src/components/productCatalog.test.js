import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import ProductCatalog from "./productCatalog";
import { MOCK_DATA } from "../mockData.js/index.js";

beforeAll(() => {
  Storage.prototype.setItem = jest.fn();
  Storage.prototype.getItem = jest.fn(() =>
    JSON.stringify({
      category: "",
      brand: "",
      minPrice: 0,
      maxPrice: 1000,
      rating: 0,
    })
  );
});

describe("ProductCatalog Component", () => {
  test("renders ProductCatalog and displays all products by default", () => {
    const { asFragment } = render(<ProductCatalog />);

    expect(asFragment()).toMatchSnapshot();

    MOCK_DATA.forEach((product) => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
    });
    expect(screen.getByText('Filter by Category')).toBeInTheDocument();
  });

  test("filters products by category", () => {
    const { asFragment } = render(<ProductCatalog />);

    expect(asFragment()).toMatchSnapshot();

    const categoryInput = screen.getByText('Electronics');
    fireEvent.change(categoryInput, { target: { value: "Electronics" } });

    const filteredProducts = MOCK_DATA.filter(
      (product) => product.category === "Electronics"
    );

    filteredProducts.forEach((product) => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
    });

    const otherProducts = MOCK_DATA.filter(
      (product) => product.category !== "Electronics"
    );
    otherProducts.forEach((product) => {
      expect(screen.getByText(product.name)).toBeInTheDocument();
    });
  });
});

afterAll(() => jest.clearAllMocks());
