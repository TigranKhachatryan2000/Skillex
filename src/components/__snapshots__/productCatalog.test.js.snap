// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCatalog Component filters products by category 1`] = `
<DocumentFragment>
  <div
    class="catalog-container"
  >
    <div
      class="filter-panel"
    >
      <div>
        <h3>
          Filter by Category
        </h3>
        <label
          for="category"
        >
          Category
        </label>
        <select
          id="category"
        >
          <option
            value=""
          >
            All
          </option>
          <option
            value="Electronics"
          >
            Electronics
          </option>
          <option
            value="Footwear"
          >
            Footwear
          </option>
          <option
            value="Clothing"
          >
            Clothing
          </option>
        </select>
        <h3>
          Filter by Brand
        </h3>
        <label
          for="brand"
        >
          Brand
        </label>
        <select
          id="brand"
        >
          <option
            value=""
          >
            All
          </option>
          <option
            value="Brand A"
          >
            Brand A
          </option>
          <option
            value="Brand B"
          >
            Brand B
          </option>
        </select>
        <h3>
          Filter by Price
        </h3>
        <input
          max="500"
          min="0"
          step="10"
          type="range"
        />
        <h3>
          Filter by Rating
        </h3>
        <input
          max="5"
          min="0"
          step="0.1"
          type="number"
          value="0"
        />
      </div>
    </div>
    <div
      class="product-list"
    >
      <div>
        <div>
          <h3>
            Wireless Headphones
          </h3>
          <p>
            $99.99
          </p>
          <p>
            4.5 Stars
          </p>
        </div>
        <div>
          <h3>
            Bluetooth Speaker
          </h3>
          <p>
            $49.99
          </p>
          <p>
            4 Stars
          </p>
        </div>
        <div>
          <h3>
            Running Shoes
          </h3>
          <p>
            $59.99
          </p>
          <p>
            4.2 Stars
          </p>
        </div>
        <div>
          <h3>
            Smartphone
          </h3>
          <p>
            $499.99
          </p>
          <p>
            4.8 Stars
          </p>
        </div>
        <div>
          <h3>
            Leather Jacket
          </h3>
          <p>
            $199.99
          </p>
          <p>
            4.7 Stars
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductCatalog Component renders ProductCatalog and displays all products by default 1`] = `
<DocumentFragment>
  <div
    class="catalog-container"
  >
    <div
      class="filter-panel"
    >
      <div>
        <h3>
          Filter by Category
        </h3>
        <label
          for="category"
        >
          Category
        </label>
        <select
          id="category"
        >
          <option
            value=""
          >
            All
          </option>
          <option
            value="Electronics"
          >
            Electronics
          </option>
          <option
            value="Footwear"
          >
            Footwear
          </option>
          <option
            value="Clothing"
          >
            Clothing
          </option>
        </select>
        <h3>
          Filter by Brand
        </h3>
        <label
          for="brand"
        >
          Brand
        </label>
        <select
          id="brand"
        >
          <option
            value=""
          >
            All
          </option>
          <option
            value="Brand A"
          >
            Brand A
          </option>
          <option
            value="Brand B"
          >
            Brand B
          </option>
        </select>
        <h3>
          Filter by Price
        </h3>
        <input
          max="500"
          min="0"
          step="10"
          type="range"
        />
        <h3>
          Filter by Rating
        </h3>
        <input
          max="5"
          min="0"
          step="0.1"
          type="number"
          value="0"
        />
      </div>
    </div>
    <div
      class="product-list"
    >
      <div>
        <div>
          <h3>
            Wireless Headphones
          </h3>
          <p>
            $99.99
          </p>
          <p>
            4.5 Stars
          </p>
        </div>
        <div>
          <h3>
            Bluetooth Speaker
          </h3>
          <p>
            $49.99
          </p>
          <p>
            4 Stars
          </p>
        </div>
        <div>
          <h3>
            Running Shoes
          </h3>
          <p>
            $59.99
          </p>
          <p>
            4.2 Stars
          </p>
        </div>
        <div>
          <h3>
            Smartphone
          </h3>
          <p>
            $499.99
          </p>
          <p>
            4.8 Stars
          </p>
        </div>
        <div>
          <h3>
            Leather Jacket
          </h3>
          <p>
            $199.99
          </p>
          <p>
            4.7 Stars
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
